File: inst/test_files/test_R6_2.js
Person2 = function(name, age) {
    // public variables and methods
    let self = this
    self.x = 10
    self.initialize = function(name, age) {
        private.name = name
        private.age = age
    }
    self.get_age = function() {
        R.print(private.age)
        R.print(private.get_age_plus_x())
    }
    // private variables and methods
    let private = {}
    private.age = null
    private.name = null
    private.get_age_plus_x = function() {
        return(R.add(private.age, self.x))
    }
    if (self.initialize) {
        self.initialize(name, age)
    }
}
elena = new Person2("Elena", 4)
elena.get_age()

File: inst/test_files/test_R6_2.R
Person2 <- R6Class("Person", public = list(
    x = 10,
   initialize = function(name, age) {
       private$name <- name
       private$age <- age
   },
   get_age = function() {
       # checks if public method can access private variable
       print(private$age)
       # checks if private method can access public variable
       print(private$get_age_plus_x())
   }
), private = list(
   age = JS_NULL,
   name = JS_NULL,
   get_age_plus_x = function() {
       return(private$age + self$x)
   }
))
elena <- Person2$new("Elena", 4)
elena$get_age()
